peer/ping_manager.go

```diff
@@ -108,6 +109,29 @@ func (m *PingManager) Start() error {
 	return err
 }
 
+// getLastRTT safely retrieves the last known RTT, returning 0 if none exists.
+func (m *PingManager) getLastRTT() time.Duration {
+	rttPtr := m.pingTime.Load()
+	if rttPtr == nil {
+		return 0
+	}
+
+	return *rttPtr
+}
+
+// getTimeWaitedOrDefault calculates the time waited since the last ping was
+// sent. If no ping was sent (pingLastSend is nil), it returns the provided
+// defaultDuration.
+func (m *PingManager) getTimeWaitedOrDefault(defaultDuration time.Duration,
```

Comment:
nit: wrap before argument instead.

Reply:
Ended up re-working things slightly to use `UnwrapOr` instead of passing in a default.